src/PlusDataCollection/WinProbe/vtkPlusWinProbeVideoSource.cxx

```diff
@@ -1199,41 +1199,111 @@ int32_t vtkPlusWinProbeVideoSource::GetSpatialCompoundCount()
 }
 
 //----------------------------------------------------------------------------
-
-void vtkPlusWinProbeVideoSource::SetBRFEnabled(bool value)
+PlusStatus vtkPlusWinProbeVideoSource::SetExtraSourceMode(Mode mode)
 {
-  if(Connected)
+  if((GetBRFEnabled() && mode == Mode::BRF) || (GetMModeEnabled() && mode == Mode::M) || (GetARFIEnabled() && mode == Mode::ARFI))
   {
-    if(m_Mode == Mode::M)
-    {
-      SetMIsEnabled(false);
-    }
-    if(m_Mode == Mode::ARFI)
-    {
-      SetARFIEnabled(false);
-    }
-    if(value)
-    {
-      SetHandleBRFInternally(false);
-      SetBFRFImageCaptureMode(2);
-    }
-    else
+    return PLUS_SUCCESS;
+  }
+  SetMIsEnabled(mode == Mode::M);
+  SetARFIIsEnabled(mode == Mode::ARFI);
+
+  if(mode == Mode::BRF)
+  {
+    SetHandleBRFInternally(false);
+    SetBFRFImageCaptureMode(2);
+  }
+  else
+  {
+    SetHandleBRFInternally(true);
+    SetBFRFImageCaptureMode(0);
+  }
+
+  if(mode == Mode::M)
+  {
+    SetMIsRevolving(m_MRevolvingEnabled);
+    SetMPRF(m_MPRF);
+    SetMAcousticLineIndex(m_MLineIndex);
+    ::SetMWidth(m_MWidth);
+    ::SetMAcousticLineCount(m_MAcousticLineCount);
+  }
+  else if(mode == Mode::ARFI)
+  {
+    m_Mode = Mode::ARFI;
+    unsigned bSize = m_SSDecimation * m_PrimaryFrameSize[1] * m_PrimaryFrameSize[0];
+    unsigned arfiDataSize = 1024 * 16 * 64 * 30;
+    unsigned timeblockSize = (4 / 2) * 64 * 30;
```

Comment:
If we want to also use the 4x beamformer variants, this formula should incorporate `quadBFCount` variable and work for both 4x and 8x variants.

Reply:
`quadBFCount` was returning 1 for me despite having an 8x engine. Does `SetARFISSQuadCount` need to be used for `GetARFISSQuadCount` to return the correct value? Would it be better to set this parameter from e.g., the DeviceConfig.xml?

Reply:
I don't know. @malixg should know.

Reply:
As this is hardware-dependent parameter, it is better not to put it in `DeviceConfig.xml`. We should query the hardware variant at run time.

Reply:
Use `GetARFIIsX8BFEnabled()` which will return a value of 1 for 8x board and a value of 0 for 4x board.